db/migrate_core_cards/20141230204340_uri_codename.rb

```diff
@@ -0,0 +1,11 @@
+# -*- encoding : utf-8 -*-
+
+class UriCodename < Wagn::Migration
+  def up
+    contentedly do
+      puts "adding new codename card for URI"
+      codename = 'URI'
+      Card.create! :type_id=>Card::CardtypeID, :name=>codename, :codename=>codename.to_name.key
```

Comment:
In my use of codename, this is the codename, just not in 'key' form.  It is also the initial name, but it may not be.   'autoname' doesn't seem the right name for this feature, maybe auto_new_name ?  Doesn't seem that related to this ticket in any case.